homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -103,45 +120,25 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 class AmcrestBinarySensor(BinarySensorEntity):
     """Binary sensor for Amcrest camera."""
 
-    def __init__(self, name, device, sensor_type):
+    def __init__(self, name, device, entity_description):
         """Initialize entity."""
-        self._name = f"{name} {BINARY_SENSORS[sensor_type][SENSOR_NAME]}"
         self._signal_name = name
         self._api = device.api
-        self._sensor_type = sensor_type
-        self._state = None
-        self._device_class = BINARY_SENSORS[sensor_type][SENSOR_DEVICE_CLASS]
-        self._event_code = BINARY_SENSORS[sensor_type][SENSOR_EVENT_CODE]
+        self.entity_description = entity_description
+        self._attr_name = f"{name} {entity_description.name}"
+        self._attr_should_poll = entity_description.key in BINARY_POLLED_SENSORS
```

Comment:
You could also add this property as a field to the custom description you already have

And change it to, e.g.,:

```suggestion
        self._attr_should_poll = entity_description.shoud_poll
```